/**
 * Copyright (c) 2014-present, The osquery authors
 *
 * This source code is licensed as defined by the LICENSE file found in the
 * root directory of this source tree.
 *
 * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
 */

syntax = "proto3";

package containerd.events;

import "google/protobuf/timestamp.proto";
import "mount.proto";

message TaskCreate {
	string container_id = 1;
	string bundle = 2;
	repeated containerd.types.Mount rootfs = 3;
	TaskIO io = 4;
	string checkpoint = 5;
	uint32 pid = 6;
}

message TaskStart {
	string container_id = 1;
	uint32 pid = 2;
}

message TaskDelete {
	string container_id = 1;
	uint32 pid = 2;
	uint32 exit_status = 3;
	google.protobuf.Timestamp exited_at = 4;
	// id is the specific exec. By default if omitted will be `""` thus matches
	// the init exec of the task matching `container_id`.
	string id = 5;
}

message TaskIO {
	string stdin = 1;
	string stdout = 2;
	string stderr = 3;
	bool terminal = 4;
}

message TaskExit {
	string container_id = 1;
	string id = 2;
	uint32 pid = 3;
	uint32 exit_status = 4;
	google.protobuf.Timestamp exited_at = 5;
}

message TaskOOM {
	string container_id = 1;
}

message TaskExecAdded {
	string container_id = 1;
	string exec_id = 2;
}

message TaskExecStarted {
	string container_id = 1;
	string exec_id = 2;
	uint32 pid = 3;
}

message TaskPaused {
	string container_id = 1;
}

message TaskResumed {
	string container_id = 1;
}

message TaskCheckpointed {
	string container_id = 1;
	string checkpoint = 2;
}